picamctl/picamctl.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash
+# akseidel 02/06/2021
+# A script automating the steps to run showmewencam's "camera-ctl".
+# The script will pause for continue or exit if run with an argument.
+
+# edit this next line to be the correct text pattern for your computer.
+portnamepat="tty.usbmodem" #typical for showmewebcam on Appl OS X
+#portnamepat="ttyACM" #probably typical for showmewebcam on Linux
+#portnamepat="tty."
+# The client is an application you want this script to also run. 
+runtheclient="true"
+clientname="Photo Booth"
+
+# initial cleanup
+initclean(){
+    reset
+    clear
+    # terminates all the ort screen sessions.
+    screen -ls | grep Attached | cut -d. -f1 | awk '{print $1}' | xargs kill
+    screen -ls | grep Detached | cut -d. -f1 | awk '{print $1}' | xargs kill
+}
+
+# run the client application
+runclient(){
+    if [ "$runtheclient" = "true" ]; then
+        # -g causes application to open in background 
+        # https://scriptingosx.com/2017/02/the-macos-open-command/
+        open -g -a "$clientname"
+    fi
+}
+
+# start the screen session
+doscreensession(){
+    echo ""
+    echo "Now establishing serial connection using 'screen' ..."
+    # For some reason screen stuffing the user, password and camera-ctl to
+    # a detached target screen does not work without the target screen having
+    # been attached at some point. Here the nest screen into a spawner screen
+    # trick is used to get around that.   
+    screen -dmS spawner
```

Comment:
One might need sudo to access the device with screen.

Reply:
I am not sure what you mean. I can tell you it took a lot of testing to determine what was going on. Apparently one cannot stuff to a detached screen that has not been attached yet. Looking online knowing that observed behaviour led to this solution. The visible, ie attached, initial screen session blocks the script until that screen is terminated. Therefore the succeeding stuffs to provide root, root, camera-ctl do not occur until the first screen is terminated. Starting the initial screen detached allowed the succeeding code to proceed, but the stuffs vanish.

Reply:
Depending on the access mode of the device, you need sudo to access it.

Reply:
Now I know what you are talking about. I would never see that on my Pi 4 where I run as root. The script now checks for root user and advises running the script with sudo. It also checks for a screen command.